
fix warnings linke: the address of '...' will always evaluate as 'true'

--- quake3/code/game/ai_dmq3.c.orig	2005-10-29 20:18:54.000000000 -0400
+++ quake3/code/game/ai_dmq3.c	2007-03-23 05:13:30.105300235 -0400
@@ -4176,7 +4176,7 @@
 		return 0;
 	}
 	trap_AAS_ValueForBSPEpairKey(ent, "classname", classname, sizeof(classname));
-	if (!classname) {
+	if (!classname[0]) {
 		BotAI_Print(PRT_ERROR, "BotGetActivateGoal: entity with model %s has no classname\n", model);
 		return 0;
 	}
--- quake3/code/tools/asm/cmdlib.c.orig	2005-10-29 20:18:48.000000000 -0400
+++ quake3/code/tools/asm/cmdlib.c	2007-03-23 05:15:09.618861102 -0400
@@ -312,7 +312,7 @@
 char *ExpandPath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!qdir[0])
 		Error ("ExpandPath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
@@ -325,7 +325,7 @@
 char *ExpandGamePath (const char *path)
 {
 	static char full[1024];
-	if (!qdir)
+	if (!qdir[0])
 		Error ("ExpandGamePath called without qdir set");
 	if (path[0] == '/' || path[0] == '\\' || path[1] == ':') {
 		strcpy( full, path );
--- quake3/code/ui/ui_players.c.orig	2005-10-29 20:18:59.000000000 -0400
+++ quake3/code/ui/ui_players.c	2007-03-23 05:14:42.364067595 -0400
@@ -1204,7 +1204,7 @@
 		}
 	}
 
-	if (headModelName && headModelName[0] == '*' ) {
+	if ( headModelName[0] == '*' ) {
 		Com_sprintf( filename, sizeof( filename ), "models/players/heads/%s/%s.md3", &headModelName[1], &headModelName[1] );
 	}
 	else {
--- quake3/code/unix/unix_main.c.orig	2006-02-26 20:39:32.000000000 -0500
+++ quake3/code/unix/unix_main.c	2007-03-23 05:02:09.592543726 -0400
@@ -798,6 +798,9 @@
   if(!libHandle && homepath)
     libHandle = try_dlopen(homepath, gamedir, fname, fqpath);
 
+  if(!libHandle)
+    libHandle = try_dlopen(QUAKELIBDIR, gamedir, fname, fqpath);
+
   if(!libHandle && basepath)
     libHandle = try_dlopen(basepath, gamedir, fname, fqpath);
 
